firstvoices/backend/views/gallery_views.py

```diff
@@ -0,0 +1,116 @@
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    create=extend_schema(
+        description="Create a new gallery for the site.",
+        responses={
+            201: OpenApiResponse(
+                description=doc_strings.success_201,
+                response=GalleryDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    update=extend_schema(
+        description="Update an existing gallery.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=GalleryDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    partial_update=extend_schema(
+        description="Update an existing gallery. Any omitted fields will be unchanged.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=GalleryDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    destroy=extend_schema(
+        description="Delete an existing gallery.",
+        responses={
+            204: OpenApiResponse(description=doc_strings.success_204_deleted),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+)
+class GalleryViewSet(SiteContentViewSetMixin, FVPermissionViewSetMixin, ModelViewSet):
+    """
+    API endpoint that allows galleries to be viewed or edited.
+    """
+
+    serializer_class = GalleryDetailSerializer
+
+    def get_queryset(self):
+        site = self.get_validated_site()
+        return (
+            Gallery.objects.filter(site__slug=site[0].slug)
+            .prefetch_related(
+                "site",
+                "site__language",
+                "created_by",
+                "last_modified_by",
+                "galleryitem_set",
+            )
+            .select_related("cover_image")
```

Comment:
Some additional prefetching of the image and thumbnail fields could be added here to ensure low response times. Feel free to send me a message if you want to connect on this.